src/go/networking/pkg/utils/pipe.go

```diff
@@ -15,14 +15,19 @@ limitations under the License.
 package utils
 
 import (
+	"context"
 	"io"
 	"net"
+	"time"
 
 	"github.com/sirupsen/logrus"
 )
 
-func Pipe(conn net.Conn, upstreamAddr string) {
-	upstream, err := net.Dial("tcp", upstreamAddr)
+func Pipe(ctx context.Context, conn net.Conn, upstreamAddr string) {
+	dialer := net.Dialer{
+		Timeout: 5 * time.Second,
```

Comment:
Do we need to set an explicit timeout? Given that we were using the default timeout (i.e. none) before… But I'm not _against_ it though.

Reply:
Yeah, I thought it wouldn't hurt to have some sort of a timeout.